source/server/listener_manager_impl.cc

```diff
@@ -281,6 +297,9 @@ void ListenerManagerImpl::drainListener(ListenerImplPtr&& listener) {
     std::lock_guard<std::mutex> guard(draining_listeners_lock_);
     draining_it = draining_listeners_.emplace(draining_listeners_.begin(), std::move(listener),
                                               workers_.size());
+    // Must be done under lock. Set is used so hot restart / multiple processes converge to a
+    // single value. Same below inside the lambda.
+    stats_.total_listeners_draining_.set(draining_listeners_.size());
```

Comment:
For my education, can you elaborate on what is meant by convergence here? I think the idea is that the new process eventually "wins" since it's the last man standing and we accept its values for these gauges?